cmd/clusterctl/cmd/version_checker.go

```diff
@@ -137,30 +146,48 @@ New clusterctl version available: v%s -> v%s
 	return "", nil
 }
 
-func (v *versionChecker) getLatestRelease(ctx context.Context) (*ReleaseInfo, error) {
+func (v *versionChecker) getLatestRelease(_ context.Context) (*ReleaseInfo, error) {
 	log := logf.Log
+
+	// Try to get latest clusterctl version number from the local state file.
+	// NOTE: local state file is ignored if older than 1d.
 	vs, err := readStateFile(v.versionFilePath)
 	if err != nil {
 		return nil, errors.Wrap(err, "unable to read version state file")
 	}
+	if vs != nil {
+		return &vs.LatestRelease, nil
+	}
 
-	// if there is no release info in the state file, pull latest release from github
-	if vs == nil {
-		release, _, err := v.githubClient.Repositories.GetLatestRelease(ctx, "kubernetes-sigs", "cluster-api")
-		if err != nil {
-			log.V(1).Info("⚠️ Unable to get latest github release for clusterctl")
-			// failing silently here so we don't error out in air-gapped
-			// environments.
-			return nil, nil //nolint:nilerr
+	// Try to get latest clusterctl version number from go modules.
+	latest, err := v.goproxyGetLatest()
+	if err != nil {
+		log.V(5).Info("error using Goproxy client to get latest versions for clusterctl, falling back to github client")
+	}
+	if latest != nil {
+		vs = &VersionState{
+			LastCheck:     time.Now(),
+			LatestRelease: *latest,
 		}
 
-		vs = &VersionState{
-			LastCheck: time.Now(),
-			LatestRelease: ReleaseInfo{
-				Version: release.GetTagName(),
-				URL:     release.GetHTMLURL(),
-			},
+		if err := writeStateFile(v.versionFilePath, vs); err != nil {
+			return nil, errors.Wrap(err, "unable to write version state file")
 		}
+		return &vs.LatestRelease, nil
+	}
+
+	// Otherwise fall back to get latest clusterctl version number from GitHub.
+	latest, err = v.gitHubGetLatest()
+	if err != nil {
+		log.V(1).Info("⚠️ Unable to get latest github release for clusterctl")
+		// failing silently here so we don't error out in air-gapped
+		// environments.
+		return nil, nil //nolint:nilerr
```

Comment:
For the air gapped scenarios, would it be worth us having a flag/env var to disable the version checking completely?

Reply:
Something like CLUSTERCTL_DISABLE_VERSIONCHECK? :)

Reply:
Ha ha, i should've checked for that. Thats a homer moment, doh!

That was a very polite/subtle response @sbueringer :)

Reply:
I just remembered that we should have something like that :)